internal/saucecloud/cloud.go

```diff
@@ -507,6 +469,53 @@ func (r *CloudRunner) remoteArchiveFiles(project interface{}, files []string, sa
 	return strings.Join(uris, ","), nil
 }
 
+// GetJUnitArtifacts retrieves the JUnit report and parent JUnit reports for the current job.
+func (r *CloudRunner) GetJUnitArtifacts(res result) (junits []report.Artifact, parentJUnits []report.Artifact) {
+	if !report.IsArtifactRequired(r.Reporters, report.JUnitArtifact) {
+		return
+	}
+
+	content, err := r.JobService.GetJobAssetFileContent(
+		context.Background(),
+		res.job.ID,
+		junit.JunitFileName,
+		res.job.IsRDC)
+	junits = append(junits, report.Artifact{
+		AssetType: report.JUnitArtifact,
+		Body:      content,
+		Error:     err,
+	})
+
+	for _, id := range res.previous {
+		jb, err := r.JobService.GetJobAssetFileContent(
+			context.Background(),
+			id,
+			junit.JunitFileName,
+			res.job.IsRDC,
+		)
+		parentJUnits = append(parentJUnits, report.Artifact{
+			AssetType: report.JUnitArtifact,
+			Body:      jb,
+			Error:     err,
+		})
+	}
+	return
+}
+
+// GetJSONArtifacts creates a list of report artifacts from the provided artifacts.
+func (r *CloudRunner) GetJSONArtifacts(files []string) (artifacts []report.Artifact) {
```

Comment:
Style: A bit of a misnomer, `Get` isn't appropriate since you aren't retrieving artifacts. You are making a transformation.

```suggestion
func (r *CloudRunner) ToJSONArtifacts(files []string) (artifacts []report.Artifact) {
```